test/cppsuite/test_harness/runtime_monitor.cxx

```diff
@@ -237,56 +253,94 @@ runtime_monitor::get_stat(scoped_cursor &cursor, int stat_field, int64_t *valuep
 }
 
 runtime_monitor::runtime_monitor(configuration *config, database &database)
-    : component("runtime_monitor", config), _postrun_stats(config), _database(database)
+    : component("runtime_monitor", config), _database(database)
 {
 }
 
 runtime_monitor::~runtime_monitor()
 {
-    for (auto &it : _stats)
-        delete it;
     _stats.clear();
```

Comment:
The call to clear() is redundant here as the destructor for  _stats will be called which will clear the _stats vector (and thereby delete all the smart pointers and thereby objects they point to). This is why smart pointers are so useful!